### PR TITLE
make it possible to disable all jvm shutdown hooks, #24077

### PR DESCRIPTION
**Pull request by patriknw**
Refs #24077

### CODE REVIEW

File: akka-actor-tests/src/test/scala/akka/actor/CoordinatedShutdownSpec.scala [hunk 1]
```diff
@@ -353,7 +353,23 @@ class CoordinatedShutdownSpec extends AkkaSpec {
         myHooksCount should ===(2) // one user, one from actor system
         cancellable.cancel()
       }
+    }
+
+    "add and remove user JVM hooks with run-by-jvm-shutdown-hook = on, akka.jvm-shutdown-hooks = on" in new JvmHookTest {
```
Comment (johanandren) [inline akka-actor-tests/src/test/scala/akka/actor/CoordinatedShutdownSpec.scala]: Test name does not match config used

File: akka-actor-tests/src/test/scala/akka/actor/CoordinatedShutdownSpec.scala [hunk 2]
```diff
@@ -366,13 +382,7 @@ class CoordinatedShutdownSpec extends AkkaSpec {
...
     def withSystemRunning(system: ActorSystem): Unit
 
-    val newSystem = ActorSystem(
-      systemName,
-      ConfigFactory.parseString(
-        """
-          akka.coordinated-shutdown.run-by-jvm-shutdown-hook = on
-          akka.coordinated-shutdown.terminate-actor-system = on
-        """))
+    val newSystem = ActorSystem(systemName, systemConfig)
```
Comment (johanandren) [inline akka-actor-tests/src/test/scala/akka/actor/CoordinatedShutdownSpec.scala]: Oh wait, so the config was actually not used in the tests!?!

**Review by johanandren — COMMENTED** [review]
LGTM with one misnamed test and a D'oh!

File: akka-actor/src/main/resources/reference.conf [hunk 3]
```diff
@@ -81,6 +81,12 @@ akka {
   # such as OutOfMemoryError
   jvm-exit-on-fatal-error = on
 
+  # Akka installs JVM shutdown hooks by default, e.g. in CoordinatedShutdown and Artery.
+  # This property makes it possible to disable all such hooks if the application itself
+  # or a higher level framework such as Play prefers to install the JVM shutdown hook and
+  # terminate the ActorSystem itself, with or without using CoordinatedShutdown.
+  jvm-shutdown-hooks = on
```
Comment (ktoso) [inline akka-actor/src/main/resources/reference.conf]: Well explained

**Review by ktoso — APPROVED** [review]
LGTM

File: akka-actor-tests/src/test/scala/akka/actor/CoordinatedShutdownSpec.scala (hunk 2 above)
Comment (patriknw) [reply akka-actor-tests/src/test/scala/akka/actor/CoordinatedShutdownSpec.scala]: yup

File: akka-actor-tests/src/test/scala/akka/actor/CoordinatedShutdownSpec.scala (hunk 1 above)
Comment (patriknw) [reply akka-actor-tests/src/test/scala/akka/actor/CoordinatedShutdownSpec.scala]: thx, fixed

**Pull request merged**